runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5875,34 +5857,24 @@ class INTERPRETER_CLASS
 				_currentThread->tempSlot = (UDATA)index;
 				rc = THROW_AIOB;
 			} else {
-				j9object_t value = NULL;
+				j9object_t value = VM_ValueTypeHelpers::loadFlattenableArrayElement(_currentThread, _objectAccessBarrier, _objectAllocate, arrayref, index, true);
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 				J9Class *arrayrefClass = J9OBJECT_CLAZZ(_currentThread, arrayref);
-				if (J9_IS_J9CLASS_FLATTENED(arrayrefClass)) {
-					j9object_t newObjectRef = _objectAllocate.inlineAllocateObject(_currentThread, ((J9ArrayClass*)arrayrefClass)->leafComponentType, false, false);
-
-					if (NULL == newObjectRef) {
-						buildGenericSpecialStackFrame(REGISTER_ARGS, 0);
-						pushObjectInSpecialFrame(REGISTER_ARGS, arrayref);
-						updateVMStruct(REGISTER_ARGS);
-						newObjectRef = _vm->memoryManagerFunctions->J9AllocateObject(_currentThread, ((J9ArrayClass*)arrayrefClass)->leafComponentType, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
-						VMStructHasBeenUpdated(REGISTER_ARGS);
-						arrayref = popObjectInSpecialFrame(REGISTER_ARGS);
-						restoreGenericSpecialStackFrame(REGISTER_ARGS);
-						if (J9_UNEXPECTED(NULL == newObjectRef)) {
-							rc = THROW_HEAP_OOM;
-							return rc;
-						}
-						arrayrefClass = VM_VMHelpers::currentClass(arrayrefClass);
+				if ((NULL == value) && J9_IS_J9CLASS_VALUETYPE(arrayrefClass)) {
```

Comment:
This is the problem, `J9_IS_J9CLASS_VALUETYPE(arrayrefClass)`, arrayclasses are never marked as VT's only the element types. This should remain as the `J9_IS_J9CLASS_FLATTENED` test